### PR TITLE
Fix some -Wfloat-conversion warnings (Pt. 2)

### PR DESCRIPTION
**Pull request by Holzhaus**
I split them up for better reviewability.

### CODE REVIEW

File: src/soundio/soundmanager.cpp [hunk 1]
```diff
@@ -319,7 +320,9 @@ void SoundManager::queryDevicesPortaudio() {
         m_devices.push_back(currentDevice);
         if (!strcmp(Pa_GetHostApiInfo(deviceInfo->hostApi)->name,
                     MIXXX_PORTAUDIO_JACK_STRING)) {
-            m_jackSampleRate = deviceInfo->defaultSampleRate;
+            m_jackSampleRate = mixxx::audio::SampleRate(
+                    static_cast<mixxx::audio::SampleRate::value_t>(
```
Comment (daschuer) [inline src/soundio/soundmanager.cpp:324]: this double cast looks odd. Can we add an overload for this type?

**Review by daschuer — CHANGES_REQUESTED** [review]
Thank you.

File: src/audio/types.h [hunk 2]
```diff
@@ -137,6 +137,10 @@ class SampleRate {
                 (m_value <= kValueMax);
     }
 
+    void operator=(const value_t value) {
```
Comment (daschuer) [inline src/audio/types.h:140]: const value_t&

**Review by daschuer — CHANGES_REQUESTED** [review]

**Review by daschuer — APPROVED** [review]
LGTM Thank you.

**Comment by ywwg** [comment]
> I split them up for better reviewability.

thanks!

**Pull request merged**